http/delete_handler.go

```diff
@@ -139,24 +142,71 @@ func (h *DeleteHandler) handleDelete(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
-func decodeDeleteRequest(ctx context.Context, r *http.Request, orgSvc influxdb.OrganizationService, bucketSvc influxdb.BucketService) (*deleteRequest, error) {
+func decodeDeleteRequest(ctx context.Context, r *http.Request, orgSvc influxdb.OrganizationService, bucketSvc influxdb.BucketService) (*deleteRequest, influxql.Expr, error) {
+	jsonErr := &errors.Error{
+		Code: errors.EInvalid,
+		Msg:  "invalid request; error parsing request json",
+	}
+
 	dr := new(deleteRequest)
-	err := json.NewDecoder(r.Body).Decode(dr)
+	buf, err := io.ReadAll(r.Body)
 	if err != nil {
-		return nil, &errors.Error{
-			Code: errors.EInvalid,
-			Msg:  "invalid request; error parsing request json",
-			Err:  err,
+		jsonErr.Err = err
+		return nil, nil, jsonErr
+	}
+	buffer := bytes.NewBuffer(buf)
+	err = json.NewDecoder(buffer).Decode(dr)
+	if err != nil {
+		jsonErr.Err = err
+		return nil, nil, jsonErr
+	}
+
+	var drd deleteRequestDecode
```

Comment:
Do you need both `dr` and `drd`?  Isn't one a superset of the other?

Reply:
So `dr` (a `deleteRequest`) when unmarshalled sets the predicate to a type that doesn't return the original string. We need that original string to do the expression parsing a few lines down. This is why I switched to the buffered read, it effectively decodes into a `deleteRequestDecode` twice, once here and once in the UnmarshalJson for the `deleteRequest` type. If we don't like that, I can remove the predicate stuff from the unmarshaling and do it elsewhere, my initial thought was to avoid that since I don't know everywhere that may be used.